components/floatingactionbutton/index.css

```diff
@@ -0,0 +1,125 @@
+  position: relative;
+
+  /* default is primary */
+  background-color: var(--highcontrast-floating-action-button-background-color, var(--mod-floating-action-button-background-color, var(--spectrum-floating-action-button-background-color)));
+
+  &:hover {
+    background-color: var(--highcontrast-floating-action-button-background-color-hover, var(--mod-floating-action-button-background-color-hover, var(--spectrum-floating-action-button-background-color-hover)));
+
+    .spectrum-FloatingActionButton-icon {
+      fill: var(--highcontrast-floating-action-button-icon-color-hover, var(--mod-floating-action-button-icon-color-hover, var(--spectrum-floating-action-button-icon-color-hover)));
+    }
+  }
+
+  &:active {
+    background-color: var(--highcontrast-floating-action-button-background-color-down, var(--mod-floating-action-button-background-color-down, var(--spectrum-floating-action-button-background-color-down)));
+
+    .spectrum-FloatingActionButton-icon {
+      fill: var(--highcontrast-floating-action-button-icon-color-down, var(--mod-floating-action-button-icon-color-down, var(--spectrum-floating-action-button-icon-color-down)));
+    }
+  }
+
+  &.is-focused,
+  &:focus, 
+  &:focus-within {
+    background-color: var(--highcontrast-floating-action-button-background-color-key-focus, var(--mod-floating-action-button-background-color-key-focus, var(--spectrum-floating-action-button-background-color-key-focus)));
+
+    .spectrum-FloatingActionButton-icon {
+      fill: var(--highcontrast-floating-action-button-icon-color-key-focus, var(--mod-floating-action-button-icon-color-key-focus, var(--spectrum-floating-action-button-icon-color-key-focus)));
+    }
+
+    &:after {
+      position: absolute;
+      inset: 0;
+      margin: calc(-1 * var(--mod-floating-action-button-focus-ring-gap, var(--spectrum-floating-action-button-focus-ring-gap)));
+      box-shadow: 0 0 0 var(--mod-floating-action-button-focus-ring-width, var(--spectrum-floating-action-button-focus-ring-width)) var(--highcontrast-floating-action-button-focus-ring-color, var(--mod-afloating-action-button-focus-ring-color, var(--spectrum-floating-action-button-focus-ring-color)));
+      border-radius: 50%;
+      pointer-events: none;
+      content: '';
+    }
+  }
+}
+
+.spectrum-Icon.spectrum-FloatingActionButton-icon {
+  block-size: var(--mod-floating-action-button-icon-size, var(--spectrum-floating-action-button-icon-size));
+  inline-size: var(--mod-floating-action-button-icon-size, var(--spectrum-floating-action-button-icon-size));
+  fill: var(--highcontrast-floating-action-button-icon-color, var(--mod-floating-action-button-icon-color, var(--spectrum-floating-action-button-icon-color)));
+}
+
+@media (forced-colors: active) {
```

Comment:
Should this @media block be closer to the top, under the other properties like they are on the other components?

Reply:
So there are a few components that have this at the bottom of the file so that it has the highest priority. I ran into that with a different component, I can't think of which one now, but I started thinking these should be at the bottom if priority was going to be an issue. Basically if you have any CSS in this media query other than the variables, it gets overridden by the CSS below it. This ensures the high contrast styling conquers all